src/uu/mktemp/src/mktemp.rs

```diff
@@ -281,7 +281,11 @@ impl Params {
             .join(prefix_from_template)
             .display()
             .to_string();
-        if options.treat_as_template && prefix.contains(MAIN_SEPARATOR) {
+
+        // Check that the prefix is valid.
+        let prefix_of_template = Path::new(prefix_from_template).to_string_lossy();
```

Comment:
Two more small things here:
1. The name is a bit confusing because it's different from `prefix_from_template` maybe those should just be the same? Or `prefix_from_template_string` or something like that?
2. Is it an `OsString`? Then we don't need to convert to a path before using `to_string_lossy()`. It could just be:
```rust
let prefix_from_template_string = prefix_from_template.to_string_lossy();
```

Reply:
prefix_from_template is &str only, so the conversion is not even necessary, it could be even simpler, I'll upload the correction:

```rs
if options.treat_as_template && prefix_from_template.contains(MAIN_SEPARATOR) {
        return Err(MkTempError::PrefixContainsDirSeparator(options.template));
}
```

So simple, but I didn't realize it in my first contact with the language. Thanks for commenting, it made me study the code and the language more to be able to comment here (I hope I'm not wrong, although the tests are ok).

Reply:
Excellent!